shm/inc/Map.hpp

```diff
@@ -0,0 +1,12 @@
+#pragma once
+
+#include "Island.hpp"
+#include <vector>
+
+class Map {
+public:
+
+private:
+    std::vector<Island&> vectorOfIslands_;
```

Comment:
Reference error in type:
In template: 'pointer' declared as a pointer to a reference of type 'Island &'clang(illegal_decl_pointer_to_reference)